x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/enrich/AbstractLookupService.java

```diff
@@ -239,7 +240,9 @@ public final void lookupAsync(R request, CancellableTask parentTask, ActionListe
 
     private void hasPrivilege(ActionListener<Void> outListener) {
         final Settings settings = clusterService.getSettings();
-        if (settings.hasValue(XPackSettings.SECURITY_ENABLED.getKey()) == false || XPackSettings.SECURITY_ENABLED.get(settings) == false) {
+        if (privilegeName == null
```

Comment:
Perhaps we should have a comment here explaining why we have null and what it means? Alternatively we could move this check to an abstract method and implement differently for ENRICH and LOOKUP JOIN.

Reply:
Made a simple abstract "getRequiredPrivilege()" method with the javadoc explaining it for null